DictionaryCompressor/FileCompressor.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace DictionaryCompressor
+{
+    internal class FileCompressor : IFileCompressor
+    {
+        public void CompressFile(string filePath, string output)
+        {
+            FileInfo file = new FileInfo(filePath);
+            using (FileStream originalFileStream = file.OpenRead())
+            {
+                if ((File.GetAttributes(file.FullName) & FileAttributes.Hidden)
+                    != FileAttributes.Hidden & file.Extension != ".cmp")
+                {
+                    output ??= file.DirectoryName;
+
+                    var compressedFileName = Path.GetFullPath(Path.Combine(output, file.Name + ".cmp"));
+                    using (FileStream compressedFileStream = File.Create(compressedFileName))
```

Comment:
Use a the `FileOpeningBuilder`

Reply:
Addressed in: https://github.com/messerli-informatik-ag/zxcvbn-cs/commit/fb7201530e2d9e5428840541795718ba6a52d009